sky/utils/controller_utils.py

```diff
@@ -380,11 +381,17 @@ def download_and_stream_latest_job_log(
         else:
             log_dir = list(log_dirs.values())[0]
             log_file = os.path.join(log_dir, 'run.log')
-
             # Print the logs to the console.
             try:
                 with open(log_file, 'r', encoding='utf-8') as f:
-                    print(f.read())
+                    # Stream the logs to the console without reading the whole
+                    # file into memory.
+                    start_streaming = False
+                    for line in f:
+                        if log_lib.LOG_FILE_START_STREAMING_AT in line:
+                            start_streaming = True
+                        if start_streaming:
+                            print(line, end='', flush=True)
```

Comment:
`f.read()` will read the whole file into the memory, while `for line in f` will only read line by line. If a user have a large log file, this will significantly reduce the memory consumption.

Reply:
Got it! Makes sense.